File: ImperatorToCK3/Source/Imperator/Provinces/ProvinceName.cpp
#include "ProvinceName.h"
#include "Log.h"
#include "ParserHelpers.h"
#include "CommonRegexes.h"



Imperator::ProvinceName::ProvinceName(std::istream& theStream) {
	registerKeys();
	parseStream(theStream);
	clearRegisteredKeywords();
}


void Imperator::ProvinceName::registerKeys() {
	registerKeyword("name", [this](std::istream& theStream) {
		name = commonItems::getString(theStream);
	});
	registerRegex(commonItems::catchallRegex, commonItems::ignoreItem);
}

File: ImperatorToCK3/Imperator/Provinces/ProvinceName.cs
using commonItems;

namespace ImperatorToCK3.Imperator.Provinces;

public sealed class ProvinceName {
	public string Name { get; private set; } = "";

	public ProvinceName(BufferedReader reader) {
		var parser = new Parser();
		RegisterKeys(parser);
		parser.ParseStream(reader);
	}
	private void RegisterKeys(Parser parser) {
		parser.RegisterKeyword("name", reader => Name = reader.GetString());
		parser.RegisterRegex(CommonRegexes.Catchall, ParserHelpers.IgnoreAndLogItem);
	}
}